app/views/dashboard/index.html.erb

```diff
@@ -5,9 +5,9 @@
 </div>
 <div class="row">
   <div class="col-8">
-    <%= render 'widget', {vms: @vms, id: 'vms_active', header: 'Active VMs', buttons: nil} %>
-    <%= render 'widget', {vms: @archivable, id: 'vms_to_be_archived', header: 'Archivable VMs', buttons: nil} %>
-    <%= render 'widget', {vms: @revivable, id: 'vms_to_be_revived', header: 'Revivable VMs', buttons: nil} %>
+    <%= render 'widget', {vms: @vms.slice(0...max_shown_vms), id: 'vms_active', header: 'Active VMs', buttons: nil} %>
+    <%= render 'widget', {vms: @archivable.slice(0...max_shown_vms), id: 'vms_to_be_archived', header: 'Archivable VMs', buttons: nil} %>
+    <%= render 'widget', {vms: @revivable.slice(0...max_shown_vms), id: 'vms_to_be_revived', header: 'Revivable VMs', buttons: nil} %>
```

Comment:
`.limit(max_shown_vms)` in `app/controllers/dashboard_controller.rb`?
See https://github.com/hpi-swt2/vm-portal/issues/443 and https://guides.rubyonrails.org/active_record_querying.html#limit-and-offset

Reply:
Vms are not a rails model but implemented by us (they wrap the vms from vSphere and are not in the database) so there is no limit method. Furthermore the problem is that we maybe do not want to show the first 10 vms out of all vms (users cannot see all vms).
There is the alternative to slice in the controller, but than there would be a third method with were vms, archivable, revivable are redefined

Reply:
Thanks for the reply.
So am I understanding it correctly, that every  API request for VMs in vSphere always returns all VMs and all filtering/slicing/limiting is done in the controllers?